pkg/network/protocols/http/ebpf_gotls.go

```diff
@@ -158,77 +252,130 @@ func supportedArch(arch string) bool {
 	return arch == string(bininspect.GoArchX86_64)
 }
 
-func (p *GoTLSProgram) handleNewBinary(binPath string) {
-	f, err := os.Open(binPath)
+func (p *GoTLSProgram) handleProcessStart(pid pid) {
+	exePath := filepath.Join(p.procRoot, strconv.FormatUint(uint64(pid), 10), "exe")
+	binPath, err := os.Readlink(exePath)
 	if err != nil {
-		log.Errorf("Could not open file %q due to %s", binPath, err)
+		log.Debugf(" could not read binary path for pid %d: %s", pid, err)
 		return
 	}
-	defer f.Close()
-	elfFile, err := elf.NewFile(f)
-	if err != nil {
-		log.Errorf("File %q could not be parsed as elf: %s", binPath, err)
+
+	var stat syscall.Stat_t
+	if err = syscall.Stat(binPath, &stat); err != nil {
```

Comment:
Do we want to check mod time to reinspect it?

Reply:
Don't we have a bigger problem actually?

Once the last process quits, we detach all hooks, meaning the next time we see it we have to redo the analysis. So, the case where checking the mod time would make sense is if there are processes from this binary still running.
But a process can still run perfectly once its binary is loaded, even if said binary is deleted or modified afterwards. Removing all the hooks to put them again would not work for the already running processes, only new ones (say, a package manager updated the binary but did not, for some reason, restart it). I am not sure how we should handle this, or if its even possible with uprobes being linked to binaries rather than processes?
Cc @guyarb